.github/workflows/cryostat-IT.yml

```diff
@@ -0,0 +1,119 @@
+on:
+  workflow_call:
+    inputs:
+      build-arch:
+        required: true
+        type: string
+      skip-itests:
+        required: false
+        type: boolean
+        default: false
+    outputs:
+      image-version:
+        description: the Cryostat application version that will be built
+        value: ${{ jobs.get-pom-properties.outputs.image-version }}
+
+permissions:
+  contents: read
```

Comment:
These permissions need revising. I think we don't need `contents: read` or `packages:read` here.

Reply:
Interestingly, artitifact-related actions are granted an "undocumented token": https://github.com/actions/upload-artifact/issues/197

Reply:
> These permissions need revising. I think we don't need contents: read or packages:read here.

Could u help check?

Reply:
I essentially copied this from the cryostat repo, so the `ci-build-image.yml` file there should also be revised/any changes made here can be copied over there too?

Reply:
Yess! Exactly! Pls do so!

Reply:
[run](https://github.com/mwangggg/cryostat-core/actions/runs/6356474255/job/17266197558) shows the permissions aren't needed... but I'm not certain if it's because it's my forked repo

Reply:
I just realized the default restrictive token already has these permissions:

https://docs.github.com/en/actions/security-guides/automatic-token-authentication

We can remove it here.